gpkit/constraints/set.py

```diff
@@ -262,3 +256,36 @@ def process_result(self, result):
         for constraint in self:
             if hasattr(constraint, "process_result"):
                 constraint.process_result(result)
+
+
+def raise_badelement(cns, i, constraint):
+    "Identify the bad element and raise a ValueError"
+    cause = "" if not isinstance(constraint, bool) else (
+        "Did the constraint list contain"
+        " an accidental equality?")
+    if len(cns) == 1:
+        loc = "as the only constraint"
+    elif i == 0:
+        loc = "at the start, before %s" % cns[i+1]
+    elif i == len(cns) - 1:
+        loc = "at the end, after %s" % cns[i-1]
+    else:
+        loc = "between %s and %s" % (cns[i-1], cns[i+1])
+    raise ValueError("%s was found %s. %s"
+                     % (type(constraint), loc, cause))
+
+
+def raise_badsubconstraintarray(constraint):
+    "Identify the bad subconstraint array and raise a ValueError"
+    cause = ("An ArrayConstraint was created with elements of"
+             " numpy.bool_")
```

Comment:
this is nitpicky, but hardcoding numpy.bool_ in the above seems ugly to me -- this could cause unexpected behavior if this method is accidentally called in another context. I'd either

a) retrieve the object type numpy.bool_ somehow instead of hardcoding it, so that it will never be incorrect, or

b) change the method name to include numpy_bool so that the method will never be used inappropriately.